### PR TITLE
[spaceship] Support individual API key

### LINKED ISSUE #21959: Individual API Key not supported
### Issue Description
Attempted to use an Individual API Key for the Account Holder to manage the Developer ID Certificate.  It is not possible to do so through a regular API Key.

##### Command executed
match with JSON key path

##### Complete output when running fastlane, including the stack trace and command used
```
[!] The request could not be completed because:
	Authentication credentials are missing or invalid. - Provide a properly configured and signed bearer token, and make sure that it has not expired. Learn more about Generating Tokens for API Requests https://developer.apple.com/go/?id=api-generating-tokens
```

### PR DESCRIPTION
**Pull request by manicmaniac**
### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [x] I've added or updated relevant unit tests.

### Motivation and Context

Resolves #21959

### Description

The main difference between a team API key and an individual API key is that the latter does not have `issuer_id`.

So I made `issuer_id` optional and if it does not exist, _fastlane_ treats the key as an individual API key.

Since the JWT token payload of an individual API key is a bit different from a team API key, I changed the logic to generate token with following the official document.

https://developer.apple.com/documentation/appstoreconnectapi/generating_tokens_for_api_requests#4313913

### Testing Steps

Sorry, I don't have a good idea to test whether if the generated JWT payload is valid or not, without exposing my individual API key.

However I tested it by writing a small lane like the following.

```ruby
# fastlane/Fastfile
lane :test do
  app_store_connect_api_key(
    key_id: ENV.fetch('INDIVIDUAL_API_KEY_ID'),
    key_content: ENV.fetch('INDIVIDUAL_API_KEY_CONTENT_BASE64'),
    is_key_content_base64: true,
    in_house: false,
    duration: 500
  )
  puts Spaceship::ConnectAPI::App.all
end
```

and confirmed that it works well in my private repo.

### CODE REVIEW

**Comment by lucgrabowski** [comment]
Thank you for your contribution :))
The code looks good to me :), I'll add a comment with an idea for a test in spec.

File: spaceship/spec/connect_api/token_spec.rb [hunk 1]
```diff
@@ -151,6 +151,21 @@
...
+      it "without issuer_id" do
+        expect(File).to receive(:binread).with('/path/to/file').and_return(private_key)
+        token = Spaceship::ConnectAPI::Token.create(
+          key_id: "key_id",
+          filepath: "/path/to/file",
+          duration: 200,
+          in_house: true
+        )
+
+        expect(token.key_id).to eq('key_id')
```
Comment (lucgrabowski) [inline spaceship/spec/connect_api/token_spec.rb:164]: Let's add a check for `issuer_id`: 
`expect(token.issuer_id).to be_nil`

File: spaceship/spec/connect_api/token_spec.rb
Comment (lucgrabowski) [inline spaceship/spec/connect_api/token_spec.rb:1]: Could you check two types of payload (with `sub` or `iss`) created for individual or team token in last context in file, 
e.g.:

```ruby
  context 'init' do
    it 'generates proper team token' do
      key = OpenSSL::PKey::EC.generate('prime256v1')
      token = Spaceship::ConnectAPI::Token.new(key_id: key_id, issuer_id: issuer_id, key: key)
      expect(token.key_id).to eq(key_id)
      expect(token.issuer_id).to eq(issuer_id)

      payload, header = JWT.decode(token.text, key, true, { algorithm: 'ES256' })

      expect(payload['iss']).to eq(issuer_id)
      expect(payload['sub']).to be_nil

      expect(payload['iat']).to be < Time.now.to_i
      expect(payload['aud']).to eq('appstoreconnect-v1')
      expect(payload['exp']).to be > Time.now.to_i

      expect(header['kid']).to eq(key_id)
      expect(header['typ']).to eq('JWT')
    end

    it 'generates proper individual token' do
      key = OpenSSL::PKey::EC.generate('prime256v1')
      token = Spaceship::ConnectAPI::Token.new(key_id: key_id, key: key)
      expect(token.key_id).to eq(key_id)
      expect(token.issuer_id).to be_nil

      payload, header = JWT.decode(token.text, key, true, { algorithm: 'ES256' })

      expect(payload['sub']).to eq('user')
      expect(payload['iss']).to be_nil

      expect(payload['iat']).to be < Time.now.to_i
      expect(payload['aud']).to eq('appstoreconnect-v1')
      expect(payload['exp']).to be > Time.now.to_i

      expect(header['kid']).to eq(key_id)
      expect(header['typ']).to eq('JWT')
    end
  end
```

File: spaceship/spec/connect_api/token_spec.rb (hunk 1 above)
Comment (manicmaniac) [reply spaceship/spec/connect_api/token_spec.rb:164]: Thank you, fixed in https://github.com/fastlane/fastlane/pull/21987/commits/69c3770d8a061766869826d83e0834f868603dbf

**Review by lucgrabowski — APPROVED** [review]
Thanks for contribution and adding tests :)

**Comment by anasyarn** [comment]
Please merge this as soon as possible, waiting for this change for a long!!!

**Comment by anasyarn** [comment]
Please merge @manicmaniac

**Comment by manicmaniac** [comment]
@anasyarn Sorry I don't have the permission to do it, just wait for the _fastlane_ team's reaction. Or you can use [my fork](https://github.com/manicmaniac/fastlane/tree/individual-api-key) until then.

**Comment by max-ott** [comment]
@anasyarn Did you test this? Would love to merge this, but as it's a rather "big" change, we need to ensure enough people from the community tested it.

**Comment by anasyarn** [comment]
No i haven’t tested it.

**Comment by dxmvsh** [comment]
Are these changes going to be released soon?

**Comment by lacett** [comment]
Any idea why commands like cert or deliver are failing using individual API keys even using this change? It raises an authentication error.

**Comment by manicmaniac** [comment]
@lacett 
AFAIK `deliver` does not rely on AppStore Connect API, so it's expected behavior.
I don't know much about `cert` but I guess it uses Provisioning API and Individual API Key has limitation to access it.

https://developer.apple.com/documentation/appstoreconnectapi/creating_api_keys_for_app_store_connect_api
> There are two types of API keys:
>
> Team
> Access to all apps, with varying levels of access based on selected roles.
>
> Individual
> Access and roles of the associated user. Individual keys aren’t able to use [Provisioning](https://developer.apple.com/documentation/appstoreconnectapi#3139092) endpoints, access [Sales and Finance](https://developer.apple.com/documentation/appstoreconnectapi/sales_and_finance), or notaryTool.

**Comment by manicmaniac** [comment]
@lucgrabowski @max-ott 

Sorry, what prevents this PR being merged?
Although it could be a rather "big" change, it still keeps backward compatibility and diff is not so big.

If we continue to wait, I cannot expect the chance that "enough" people test this one 🤔

**Review by joshdholtz — APPROVED** [review]
This looks good to me! Thank you for adding this ❤

**Pull request merged**